// @lens-protocol/api-bindings/mocks/dist/lens-protocol-api-bindings-mocks.esm.js
import { hn as _arrayLikeToArray, a as _unsupportedIterableToArray, S as SafeApolloClient, k as createLensCache, A as ApolloServerErrorCode, _ as _objectSpread2, T as TxStatus, aC as TriStateValue, ae as ProfileActionHistoryType, ao as PublicationReactionType, a5 as ModuleType, ci as FragmentPaginatedResultInfo, di as CreateActOnOpenActionTypedDataDocument, dg as ActOnOpenActionDocument, dx as RevokeAuthenticationDocument, e_ as CreateOnchainSetProfileMetadataTypedDataDocument, eS as UnblockDocument, f4 as CreateUnblockProfilesTypedDataDocument, eQ as BlockDocument, f2 as CreateBlockProfilesTypedDataDocument, f6 as CreateFollowTypedDataDocument, eU as FollowDocument, f8 as CreateUnfollowTypedDataDocument, eW as UnfollowDocument, fc as CreateSetFollowModuleTypedDataDocument, fa as SetFollowModuleDocument, fi as CreateLinkHandleToProfileTypedDataDocument, fe as LinkHandleToProfileDocument, fk as CreateUnlinkHandleFromProfileTypedDataDocument, fg as UnlinkHandleFromProfileDocument, gb as HidePublicationDocument, gd as ReportPublicationDocument, gx as AddReactionDocument, gz as RemoveReactionDocument, fH as CreateOnchainPostTypedDataDocument, fX as PostOnchainDocument, fP as CreateMomokaPostTypedDataDocument, g3 as PostOnMomokaDocument, fJ as CreateOnchainCommentTypedDataDocument, fZ as CommentOnchainDocument, fR as CreateMomokaCommentTypedDataDocument, g5 as CommentOnMomokaDocument, fT as CreateMomokaMirrorTypedDataDocument, g7 as MirrorOnMomokaDocument, fL as CreateOnchainMirrorTypedDataDocument, f$ as MirrorOnchainDocument, fN as CreateOnchainQuoteTypedDataDocument, g1 as QuoteOnchainDocument, fV as CreateMomokaQuoteTypedDataDocument, g9 as QuoteOnMomokaDocument, gl as AddPublicationBookmarkDocument, gn as RemovePublicationBookmarkDocument, gh as CreateLegacyCollectTypedDataDocument, gf as LegacyCollectDocument, gp as AddPublicationNotInterestedDocument, gr as UndoPublicationNotInterestedDocument, h0 as BroadcastOnchainDocument, h2 as BroadcastOnMomokaDocument, dz as ExplorePublicationsDocument, dF as FeedDocument, dI as FeedHighlightsDocument, dC as ExploreProfilesDocument, e2 as CurrenciesDocument, e5 as ModuleMetadataDocument, e8 as NotificationsDocument, eC as ProfileActionHistoryDocument, eb as ProfileDocument, ee as ProfilesDocument, et as MutualFollowersDocument, eq as FollowersDocument, en as FollowingDocument, gQ as SearchProfilesDocument, ek as ProfileRecommendationsDocument, ew as WhoActedOnPublicationDocument, gB as WhoReactedPublicationDocument, fs as PublicationDocument, fv as PublicationsDocument, gN as SearchPublicationsDocument, fE as PublicationBookmarksDocument, gK as FollowRevenuesDocument, gH as RevenueFromPublicationDocument, gE as RevenueFromPublicationsDocument, gZ as LensTransactionStatusDocument, $ as LensTransactionStatusType, h4 as GenerateLensApiRelayAddressDocument } from '../../dist/unsupportedIterableToArray-56a65d61.esm.js';
import { mockSingleLink } from '@apollo/client/testing';
import { GraphQLError } from 'graphql';
import { faker } from '@faker-js/faker';
import { mockCreatePostRequest, mockProfileId, mockTransactionHash, mockPublicationId, mockNonce } from '@lens-protocol/domain/mocks';
import { ChainType } from '@lens-protocol/shared-kernel';
import { mockEvmAddress } from '@lens-protocol/shared-kernel/mocks';
import { mock } from 'jest-mock-extended';
import '@apollo/client';
import '@apollo/client/utilities';
import '@lens-protocol/domain/entities';
import '@lens-protocol/domain/use-cases/publications';

function _arrayWithoutHoles(arr) {
  if (Array.isArray(arr)) return _arrayLikeToArray(arr);
}

function _iterableToArray(iter) {
  if (typeof Symbol !== "undefined" && iter[Symbol.iterator] != null || iter["@@iterator"] != null) return Array.from(iter);
}

function _nonIterableSpread() {
  throw new TypeError("Invalid attempt to spread non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}

function _toConsumableArray(arr) {
  return _arrayWithoutHoles(arr) || _iterableToArray(arr) || _unsupportedIterableToArray(arr) || _nonIterableSpread();
}

function mockLensApolloClient() {
  var mocks = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : [];
  return new SafeApolloClient({
    cache: createLensCache(),
    link: mockSingleLink.apply(void 0, _toConsumableArray(mocks)).setOnError(function (error) {
      throw error;
    }),
    pollingInterval: 1 // FAST
  });
}
function createGraphQLError(_ref) {
  var code = _ref.code,
    _ref$message = _ref.message,
    message = _ref$message === void 0 ? 'No pings please!' : _ref$message;
  return new GraphQLError(message, {
    extensions: {
      code: code
    }
  });
}
function createValidationGraphQLError() {
  var message = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : 'No pings please!';
  // why GraphQLError and not ApolloError? Cause GraphQLError responses are rehydrated by ApolloClient
  // into ApolloError instances, and this dummy error is used to test the error logic of SafeApolloClient
  // which has a proper HttpLink instance (instead of mockSingleLink) that will rehydrate the error.
  return createGraphQLError({
    message: message,
    code: ApolloServerErrorCode.GRAPHQL_VALIDATION_FAILED
  });
}
function mockValidationErrorResponse(document) {
  return {
    request: {
      query: document
    },
    result: {
      errors: [createValidationGraphQLError()]
    }
  };
}
function mockGenericSuccessResponse(document, data) {
  return {
    request: {
      query: document
    },
    result: {
      data: data
    }
  };
}
function mockGenericErrorResponse(document) {
  return {
    request: {
      query: document
    },
    result: {
      errors: [createGraphQLError({
        message: 'No pings please!',
        code: ApolloServerErrorCode.INTERNAL_SERVER_ERROR
      })]
    }
  };
}
function createHttpJsonResponse(status, body) {
  var headers = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : {};
  return new Response(JSON.stringify(body), {
    status: status,
    headers: _objectSpread2(_objectSpread2({}, headers), {}, {
      'Content-Type': 'application/json'
    })
  });
}
function createUnauthenticatedHttpResponse() {
  return createHttpJsonResponse(401, {
    data: null,
    errors: [createGraphQLError({
      message: 'Authentication required',
      code: 'UNAUTHENTICATED'
    })]
  });
}

function mockLensCache() {
  return createLensCache();
}
function mockTransactionState(partial) {
  return _objectSpread2({
    id: faker.datatype.uuid(),
    status: TxStatus.PENDING,
    request: mockCreatePostRequest()
  }, partial);
}

function mockNetworkAddressFragment(overrides) {
  return _objectSpread2(_objectSpread2({
    address: mockEvmAddress(),
    chainId: 1
  }, overrides), {}, {
    __typename: 'NetworkAddress'
  });
}
function mockErc20FragmentFrom(erc20) {
  return {
    __typename: 'Erc20',
    name: erc20.name,
    symbol: erc20.symbol,
    decimals: erc20.decimals,
    contract: mockNetworkAddressFragment({
      address: erc20.address,
      chainId: erc20.chainType === ChainType.ETHEREUM ? 1 : 137
    })
  };
}
function mockAmountFragmentFrom(amount) {
  return {
    __typename: 'Amount',
    asset: mockErc20FragmentFrom(amount.asset),
    value: amount.toSignificantDigits(),
    rate: null
  };
}
function mockOptimisticStatusResultFragment(overrides) {
  return _objectSpread2(_objectSpread2({
    value: true,
    isFinalisedOnchain: false
  }, overrides), {}, {
    __typename: 'OptimisticStatusResult'
  });
}
function mockProfileOperationsFragment(overrides) {
  return _objectSpread2(_objectSpread2({
    id: mockProfileId(),
    canBlock: false,
    canUnblock: false,
    canFollow: TriStateValue.Unknown,
    canUnfollow: false,
    isBlockedByMe: mockOptimisticStatusResultFragment(),
    isFollowedByMe: mockOptimisticStatusResultFragment(),
    isFollowingMe: mockOptimisticStatusResultFragment()
  }, overrides), {}, {
    __typename: 'ProfileOperations'
  });
}
function mockProfileOnchainIdentityFragment(overrides) {
  return _objectSpread2(_objectSpread2({
    proofOfHumanity: false,
    ens: null,
    sybilDotOrg: {
      verified: true,
      source: {
        twitter: {
          handle: null
        }
      }
    },
    worldcoin: {
      isHuman: false
    }
  }, overrides), {}, {
    __typename: 'ProfileOnchainIdentity'
  });
}
function mockHandleInfo() {
  var localName = faker.internet.userName();
  var namespace = faker.internet.domainWord();
  return {
    id: "".concat(namespace, "/").concat(localName, "}"),
    fullHandle: "".concat(localName, "/").concat(namespace),
    namespace: namespace,
    localName: localName,
    ownedBy: mockEvmAddress(),
    suggestedFormatted: {
      full: "".concat(namespace, "/@").concat(localName),
      localName: "@".concat(localName)
    },
    linkedTo: {
      nftTokenId: '1',
      contract: mockNetworkAddressFragment()
    },
    __typename: 'HandleInfo'
  };
}
function mockProfileFragment(overrides) {
  return _objectSpread2(_objectSpread2({
    id: mockProfileId(),
    txHash: mockTransactionHash(),
    createdAt: faker.date.past().toISOString(),
    interests: [],
    handle: mockHandleInfo(),
    invitesLeft: 0,
    sponsor: false,
    signless: false,
    ownedBy: mockNetworkAddressFragment(),
    operations: mockProfileOperationsFragment(),
    guardian: null,
    onchainIdentity: mockProfileOnchainIdentityFragment(),
    followNftAddress: mockNetworkAddressFragment(),
    followModule: null,
    metadata: null,
    invitedBy: null,
    stats: mockProfileStatsFragment(),
    globalStats: mockProfileStatsFragment(),
    peerToPeerRecommendedByMe: false
  }, overrides), {}, {
    __typename: 'Profile'
  });
}
function mockPostFragment(overrides) {
  var publicationId = mockPublicationId();
  var stats = mockPublicationStatsFragment({
    id: publicationId
  });
  return _objectSpread2(_objectSpread2({
    id: publicationId,
    isHidden: false,
    txHash: mockTransactionHash(),
    by: mockProfileFragment(),
    createdAt: faker.date.past().toISOString(),
    publishedOn: null,
    momoka: null,
    operations: mockPublicationOperationsFragment(),
    metadata: mockPublicationTextOnlyMetadata(),
    openActionModules: [],
    referenceModule: null,
    stats: stats,
    globalStats: stats,
    isEncrypted: false,
    hashtagsMentioned: [],
    profilesMentioned: []
  }, overrides), {}, {
    __typename: 'Post'
  });
}
function mockCommentFragment(overrides) {
  var mainPost = mockPostFragment();
  return _objectSpread2(_objectSpread2({
    id: mockPublicationId(),
    isHidden: false,
    hiddenByAuthor: false,
    txHash: mockTransactionHash(),
    by: mockProfileFragment(),
    createdAt: faker.date.past().toISOString(),
    publishedOn: null,
    momoka: null,
    operations: mockPublicationOperationsFragment(),
    metadata: mockPublicationTextOnlyMetadata(),
    openActionModules: [],
    referenceModule: null,
    root: mainPost,
    commentOn: mainPost,
    firstComment: null,
    stats: mockPublicationStatsFragment(),
    globalStats: mockPublicationStatsFragment(),
    isEncrypted: false,
    hashtagsMentioned: [],
    profilesMentioned: []
  }, overrides), {}, {
    __typename: 'Comment'
  });
}
function mockQuoteFragment(overrides) {
  return _objectSpread2(_objectSpread2({
    id: mockPublicationId(),
    isHidden: false,
    txHash: mockTransactionHash(),
    by: mockProfileFragment(),
    createdAt: faker.date.past().toISOString(),
    publishedOn: null,
    momoka: null,
    operations: mockPublicationOperationsFragment(),
    metadata: mockPublicationTextOnlyMetadata(),
    openActionModules: [],
    referenceModule: null,
    quoteOn: mockPostFragment(),
    stats: mockPublicationStatsFragment(),
    globalStats: mockPublicationStatsFragment(),
    isEncrypted: false,
    hashtagsMentioned: [],
    profilesMentioned: []
  }, overrides), {}, {
    __typename: 'Quote'
  });
}
function mockUnknownReferenceModuleSettings(overrides) {
  return _objectSpread2(_objectSpread2({
    contract: mockNetworkAddressFragment(),
    signlessApproved: true,
    sponsoredApproved: true,
    initializeCalldata: null,
    initializeResultData: null,
    verified: false
  }, overrides), {}, {
    __typename: 'UnknownReferenceModuleSettings'
  });
}
function mockMirrorFragment(overrides) {
  var mainPost = mockPostFragment();
  return _objectSpread2(_objectSpread2({
    id: mockPublicationId(),
    isHidden: false,
    txHash: mockTransactionHash(),
    by: mockProfileFragment(),
    createdAt: faker.date.past().toISOString(),
    publishedOn: null,
    momoka: null,
    mirrorOn: mainPost
  }, overrides), {}, {
    __typename: 'Mirror'
  });
}
function mockPublicationTextOnlyMetadata() {
  var overrides = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};
  return _objectSpread2(_objectSpread2({
    id: faker.helpers.unique(faker.datatype.uuid),
    rawURI: faker.internet.url(),
    locale: 'en',
    tags: null,
    contentWarning: null,
    hideFromFeed: false,
    appId: null,
    content: faker.lorem.words(5),
    marketplace: null,
    attributes: null,
    encryptedWith: null
  }, overrides), {}, {
    __typename: 'TextOnlyMetadataV3'
  });
}
function mockCanDecryptResponseFragment(overrides) {
  return _objectSpread2(_objectSpread2({
    result: false,
    reasons: null,
    extraDetails: null
  }, overrides), {}, {
    __typename: 'CanDecryptResponse'
  });
}
function mockPublicationOperationsFragment() {
  var overrides = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};
  return _objectSpread2(_objectSpread2({
    id: mockPublicationId(),
    isNotInterested: false,
    hasBookmarked: false,
    hasReported: false,
    canCollect: TriStateValue.Unknown,
    canComment: TriStateValue.Unknown,
    canMirror: TriStateValue.Unknown,
    canQuote: TriStateValue.Unknown,
    hasMirrored: false,
    hasUpvoted: false,
    hasDownvoted: false,
    hasCollected: mockOptimisticStatusResultFragment(),
    canDecrypt: mockCanDecryptResponseFragment()
  }, overrides), {}, {
    __typename: 'PublicationOperations'
  });
}
function mockPaginatedResultInfo() {
  var overrides = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};
  return _objectSpread2(_objectSpread2({
    prev: null,
    next: null
  }, overrides), {}, {
    __typename: 'PaginatedResultInfo'
  });
}
function mockPublicationStatsFragment() {
  var overrides = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};
  return _objectSpread2(_objectSpread2({
    id: mockPublicationId(),
    comments: faker.datatype.number(),
    mirrors: faker.datatype.number(),
    quotes: faker.datatype.number(),
    bookmarks: faker.datatype.number(),
    collects: faker.datatype.number(),
    upvotes: faker.datatype.number(),
    downvotes: faker.datatype.number()
  }, overrides), {}, {
    __typename: 'PublicationStats'
  });
}
function mockProfileStatsFragment() {
  var overrides = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};
  return _objectSpread2(_objectSpread2({
    id: mockProfileId(),
    followers: faker.datatype.number(),
    following: faker.datatype.number(),
    comments: faker.datatype.number(),
    posts: faker.datatype.number(),
    mirrors: faker.datatype.number(),
    quotes: faker.datatype.number(),
    publications: faker.datatype.number(),
    collects: faker.datatype.number(),
    upvotes: faker.datatype.number(),
    downvotes: faker.datatype.number(),
    upvoted: faker.datatype.number(),
    downvoted: faker.datatype.number(),
    lensClassifierScore: 0
  }, overrides), {}, {
    __typename: 'ProfileStats'
  });
}
function mockFeedItemFragment(overrides) {
  return _objectSpread2(_objectSpread2({
    id: faker.datatype.uuid(),
    root: mockPostFragment(),
    comments: [],
    mirrors: [],
    reactions: []
  }, overrides), {}, {
    __typename: 'FeedItem'
  });
}
function mockProfileActionHistoryFragment() {
  var overrides = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};
  return _objectSpread2(_objectSpread2({
    id: faker.datatype.number(),
    actionType: ProfileActionHistoryType.LoggedIn,
    who: mockEvmAddress(),
    txHash: mockTransactionHash(),
    actionedOn: faker.date.past().toISOString()
  }, overrides), {}, {
    __typename: 'ProfileActionHistory'
  });
}
function mockMentionNotification(overrides) {
  return _objectSpread2({
    __typename: 'MentionNotification',
    id: faker.datatype.uuid(),
    publication: mockPostFragment()
  }, overrides);
}
function mockProfileReactionResultFragment() {
  var overrides = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};
  return _objectSpread2(_objectSpread2({
    reaction: PublicationReactionType.Upvote,
    reactionAt: faker.date.past().toISOString()
  }, overrides), {}, {
    __typename: 'ProfileReactionResult'
  });
}
function mockProfileWhoReactedResultFragment() {
  var overrides = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};
  return _objectSpread2(_objectSpread2({
    profile: mockProfileFragment(),
    reactions: [mockProfileReactionResultFragment()]
  }, overrides), {}, {
    __typename: 'ProfileWhoReactedResult'
  });
}
function mockErc20Fragment() {
  var overrides = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};
  return _objectSpread2(_objectSpread2({
    name: 'Wrapped MATIC',
    symbol: 'WMATIC',
    decimals: 18,
    contract: mockNetworkAddressFragment()
  }, overrides), {}, {
    __typename: 'Erc20'
  });
}
function mockAmountFragment() {
  var overrides = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};
  return _objectSpread2(_objectSpread2({
    value: faker.datatype.number().toString(),
    asset: mockErc20Fragment(),
    rate: null
  }, overrides), {}, {
    __typename: 'Amount'
  });
}
function mockRevenueAggregateFragment() {
  var overrides = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};
  return _objectSpread2(_objectSpread2({
    total: mockAmountFragment()
  }, overrides), {}, {
    __typename: 'RevenueAggregate'
  });
}
function mockPublicationRevenueFragment() {
  var _ref = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {},
    _ref$publication = _ref.publication,
    publication = _ref$publication === void 0 ? mockPostFragment() : _ref$publication;
  return {
    __typename: 'PublicationRevenue',
    publication: publication,
    revenue: [mockRevenueAggregateFragment()]
  };
}
function mockFeeFollowModuleSettingsFragment(overrides) {
  return mock(_objectSpread2(_objectSpread2({}, overrides), {}, {
    __typename: 'FeeFollowModuleSettings'
  }));
}
function mockSimpleCollectOpenActionSettingsFragment(overrides) {
  return mock(_objectSpread2(_objectSpread2({}, overrides), {}, {
    __typename: 'SimpleCollectOpenActionSettings'
  }));
}
function mockMultirecipientFeeCollectOpenActionSettingsFragment(overrides) {
  return mock(_objectSpread2(_objectSpread2({}, overrides), {}, {
    __typename: 'MultirecipientFeeCollectOpenActionSettings'
  }));
}
function mockUnknownOpenActionModuleSettingsFragment(overrides) {
  return mock(_objectSpread2(_objectSpread2({
    contract: mockNetworkAddressFragment()
  }, overrides), {}, {
    __typename: 'UnknownOpenActionModuleSettings'
  }));
}
function mockLegacyFreeCollectModuleSettingsFragment(overrides) {
  return mock(_objectSpread2(_objectSpread2({}, overrides), {}, {
    __typename: 'LegacyFreeCollectModuleSettings'
  }));
}
function mockLegacyFeeCollectModuleSettingsFragment(overrides) {
  return mock(_objectSpread2(_objectSpread2({}, overrides), {}, {
    __typename: 'LegacyFeeCollectModuleSettings'
  }));
}
function mockLegacyLimitedFeeCollectModuleSettingsFragment(overrides) {
  return mock(_objectSpread2(_objectSpread2({}, overrides), {}, {
    __typename: 'LegacyLimitedFeeCollectModuleSettings'
  }));
}
function mockLegacyLimitedTimedFeeCollectModuleSettingsFragment(overrides) {
  return mock(_objectSpread2(_objectSpread2({}, overrides), {}, {
    __typename: 'LegacyLimitedTimedFeeCollectModuleSettings'
  }));
}
function mockLegacyRevertCollectModuleSettingsFragment(overrides) {
  return mock(_objectSpread2(_objectSpread2({}, overrides), {}, {
    __typename: 'LegacyRevertCollectModuleSettings'
  }));
}
function mockLegacyTimedFeeCollectModuleSettingsFragment(overrides) {
  return mock(_objectSpread2(_objectSpread2({}, overrides), {}, {
    __typename: 'LegacyTimedFeeCollectModuleSettings'
  }));
}
function mockLegacyMultirecipientFeeCollectModuleSettingsFragment(overrides) {
  return mock(_objectSpread2(_objectSpread2({}, overrides), {}, {
    __typename: 'LegacyMultirecipientFeeCollectModuleSettings'
  }));
}
function mockLegacySimpleCollectModuleSettingsFragment(overrides) {
  return mock(_objectSpread2(_objectSpread2({}, overrides), {}, {
    __typename: 'LegacySimpleCollectModuleSettings'
  }));
}
function mockLegacyErc4626FeeCollectModuleSettingsFragment(overrides) {
  return mock(_objectSpread2(_objectSpread2({}, overrides), {}, {
    __typename: 'LegacyERC4626FeeCollectModuleSettings'
  }));
}
function mockLegacyAaveFeeCollectModuleSettingsFragment(overrides) {
  return mock(_objectSpread2(_objectSpread2({}, overrides), {}, {
    __typename: 'LegacyAaveFeeCollectModuleSettings'
  }));
}
function mockModuleMetadataFragment(overrides) {
  return _objectSpread2({
    __typename: 'ModuleMetadata',
    authors: [],
    description: '',
    initializeCalldataABI: '[]',
    initializeResultDataABI: null,
    name: '',
    processCalldataABI: '[]',
    title: '',
    attributes: []
  }, overrides);
}
function mockModuleMetadataResultFragment(overrides) {
  return _objectSpread2(_objectSpread2({
    metadata: mockModuleMetadataFragment(),
    moduleType: ModuleType.Follow,
    signlessApproved: true,
    sponsoredApproved: true,
    verified: true
  }, overrides), {}, {
    __typename: 'GetModuleMetadataResult'
  });
}

/**
 * AST for:
 *
 * ```graphql
 * query GetHero($cursor: String!) {
 *   result: hero(cursor: $cursor) {
 *     items {
 *       name
 *     }
 *     pageInfo {
 *       ...PaginatedResultInfo
 *     }
 *   }
 * }
 * ```
 *
 * We couldn't use `graphql-tag` because it's does rely on fragment `DocumentNode` to
 * have `loc` property (from `gql` usage itself), which is not the case for our generated
 * fragments.
 */

var AnyPaginatedQueryDocument = {
  kind: 'Document',
  definitions: [{
    kind: 'OperationDefinition',
    operation: 'query',
    name: {
      kind: 'Name',
      value: 'GetHero'
    },
    variableDefinitions: [{
      kind: 'VariableDefinition',
      variable: {
        kind: 'Variable',
        name: {
          kind: 'Name',
          value: 'cursor'
        }
      },
      type: {
        kind: 'NamedType',
        name: {
          kind: 'Name',
          value: 'Cursor'
        }
      }
    }],
    selectionSet: {
      kind: 'SelectionSet',
      selections: [{
        kind: 'Field',
        alias: {
          kind: 'Name',
          value: 'result'
        },
        name: {
          kind: 'Name',
          value: 'hero'
        },
        arguments: [{
          kind: 'Argument',
          name: {
            kind: 'Name',
            value: 'cursor'
          },
          value: {
            kind: 'Variable',
            name: {
              kind: 'Name',
              value: 'cursor'
            }
          }
        }],
        selectionSet: {
          kind: 'SelectionSet',
          selections: [{
            kind: 'Field',
            name: {
              kind: 'Name',
              value: 'items'
            },
            selectionSet: {
              kind: 'SelectionSet',
              selections: [{
                kind: 'Field',
                name: {
                  kind: 'Name',
                  value: 'name'
                }
              }]
            }
          }, {
            kind: 'Field',
            name: {
              kind: 'Name',
              value: 'pageInfo'
            },
            selectionSet: {
              kind: 'SelectionSet',
              selections: [{
                kind: 'FragmentSpread',
                name: {
                  kind: 'Name',
                  value: 'PaginatedResultInfo'
                }
              }]
            }
          }]
        }
      }]
    }
  }].concat(_toConsumableArray(FragmentPaginatedResultInfo.definitions))
};
function mockAnyPaginatedItem() {
  return {
    name: faker.helpers.arrayElement(['Luke Skywalker', 'Darth Vader', 'Han Solo'])
  };
}
function mockAnyPaginatedQueryResult() {
  var _ref = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {},
    _ref$items = _ref.items,
    items = _ref$items === void 0 ? [mockAnyPaginatedItem()] : _ref$items,
    _ref$next = _ref.next,
    next = _ref$next === void 0 ? null : _ref$next,
    _ref$prev = _ref.prev,
    prev = _ref$prev === void 0 ? null : _ref$prev;
  return {
    items: items,
    pageInfo: mockPaginatedResultInfo({
      next: next,
      prev: prev
    })
  };
}
function mockAnyPaginatedQueryResponse(_ref2) {
  var _ref2$variables = _ref2.variables,
    variables = _ref2$variables === void 0 ? {} : _ref2$variables,
    result = _ref2.result;
  return {
    request: {
      query: AnyPaginatedQueryDocument,
      variables: variables
    },
    result: {
      data: {
        result: result
      }
    }
  };
}

function mockEIP712TypedDataField() {
  return {
    name: 'nonce',
    type: 'uint256'
  };
}
function mockEIP712TypedDataDomain() {
  return {
    name: 'Lens',
    version: '1',
    chainId: 0,
    verifyingContract: mockEvmAddress()
  };
}
function mockCreateTypedDataResult(__typename, typedData) {
  return {
    __typename: __typename,
    id: faker.datatype.uuid(),
    expiresAt: faker.date.future().toISOString(),
    typedData: typedData
  };
}

function mockCreateActOnOpenActionTypedDataData() {
  var _ref = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {},
    _ref$nonce = _ref.nonce,
    nonce = _ref$nonce === void 0 ? mockNonce() : _ref$nonce;
  return {
    result: mockCreateTypedDataResult('CreateActOnOpenActionBroadcastItemResult', {
      types: {
        Act: [mockEIP712TypedDataField()]
      },
      domain: mockEIP712TypedDataDomain(),
      message: {
        nonce: nonce,
        deadline: 1644303500,
        publicationActedProfileId: mockProfileId(),
        publicationActedId: '0x01',
        actorProfileId: mockProfileId(),
        referrerProfileIds: [],
        referrerPubIds: [],
        actionModuleAddress: mockEvmAddress(),
        actionModuleData: '0x'
      }
    })
  };
}
function mockCreateActOnOpenActionTypedDataResponse(_ref2) {
  var variables = _ref2.variables,
    data = _ref2.data;
  return {
    request: {
      query: CreateActOnOpenActionTypedDataDocument,
      variables: variables
    },
    result: {
      data: data
    }
  };
}
function mockActOnOpenActionResponse(_ref3) {
  var variables = _ref3.variables,
    data = _ref3.data;
  return {
    request: {
      query: ActOnOpenActionDocument,
      variables: variables
    },
    result: {
      data: data
    }
  };
}

function mockRevokeAuthenticationResponse(variables) {
  return {
    request: {
      query: RevokeAuthenticationDocument,
      variables: variables
    },
    result: {
      data: {
        revokeAuthentication: null
      }
    }
  };
}

function mockCreateSetProfileMetadataTypedDataData() {
  var _ref = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {},
    _ref$metadataURI = _ref.metadataURI,
    metadataURI = _ref$metadataURI === void 0 ? faker.internet.url() : _ref$metadataURI,
    _ref$nonce = _ref.nonce,
    nonce = _ref$nonce === void 0 ? mockNonce() : _ref$nonce;
  return {
    result: mockCreateTypedDataResult('CreateOnchainSetProfileMetadataBroadcastItemResult', {
      __typename: 'CreateOnchainSetProfileMetadataBroadcastItemResult',
      types: {
        __typename: 'CreateSetProfileMetadataURIEIP712TypedDataTypes',
        SetProfileMetadataURI: [mockEIP712TypedDataField()]
      },
      domain: mockEIP712TypedDataDomain(),
      message: {
        __typename: 'CreateSetProfileMetadataURIEIP712TypedDataValue',
        nonce: nonce,
        deadline: faker.datatype.number(),
        profileId: mockProfileId(),
        metadataURI: metadataURI
      }
    })
  };
}
function mockCreateSetProfileMetadataTypedDataResponse(_ref2) {
  var request = _ref2.request,
    overrideSigNonce = _ref2.overrideSigNonce,
    _ref2$data = _ref2.data,
    data = _ref2$data === void 0 ? mockCreateSetProfileMetadataTypedDataData({
      nonce: overrideSigNonce
    }) : _ref2$data;
  return {
    request: {
      query: CreateOnchainSetProfileMetadataTypedDataDocument,
      variables: {
        request: request,
        options: overrideSigNonce ? {
          overrideSigNonce: overrideSigNonce
        } : undefined
      }
    },
    result: {
      data: data
    }
  };
}
function mockUnblockProfilesTypedDataData() {
  var _ref3 = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {},
    _ref3$nonce = _ref3.nonce,
    nonce = _ref3$nonce === void 0 ? mockNonce() : _ref3$nonce;
  return {
    result: mockCreateTypedDataResult('CreateUnblockProfilesBroadcastItemResult', {
      types: {
        SetBlockStatus: [mockEIP712TypedDataField()]
      },
      domain: mockEIP712TypedDataDomain(),
      message: {
        nonce: nonce,
        deadline: 1644303500,
        blockStatus: [false],
        byProfileId: mockProfileId(),
        idsOfProfilesToSetBlockStatus: [mockProfileId()],
        datas: []
      }
    })
  };
}
function mockUnblockProfilesResponse(_ref4) {
  var variables = _ref4.variables,
    data = _ref4.data;
  return {
    request: {
      query: UnblockDocument,
      variables: variables
    },
    result: {
      data: data
    }
  };
}
function mockUnblockProfilesTypedDataResponse(_ref5) {
  var variables = _ref5.variables,
    data = _ref5.data;
  return {
    request: {
      query: CreateUnblockProfilesTypedDataDocument,
      variables: variables
    },
    result: {
      data: data
    }
  };
}
function mockBlockProfilesResponse(_ref6) {
  var variables = _ref6.variables,
    data = _ref6.data;
  return {
    request: {
      query: BlockDocument,
      variables: variables
    },
    result: {
      data: data
    }
  };
}
function mockBlockProfilesTypedDataData() {
  var _ref7 = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {},
    _ref7$nonce = _ref7.nonce,
    nonce = _ref7$nonce === void 0 ? mockNonce() : _ref7$nonce;
  return {
    result: mockCreateTypedDataResult('CreateBlockProfilesBroadcastItemResult', {
      types: {
        SetBlockStatus: [mockEIP712TypedDataField()]
      },
      domain: mockEIP712TypedDataDomain(),
      message: {
        nonce: nonce,
        deadline: 1644303500,
        blockStatus: [true],
        byProfileId: mockProfileId(),
        idsOfProfilesToSetBlockStatus: [mockProfileId()],
        datas: []
      }
    })
  };
}
function mockBlockProfilesTypedDataResponse(_ref8) {
  var variables = _ref8.variables,
    data = _ref8.data;
  return {
    request: {
      query: CreateBlockProfilesTypedDataDocument,
      variables: variables
    },
    result: {
      data: data
    }
  };
}
function mockCreateFollowTypedDataData() {
  var _ref9 = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {},
    _ref9$nonce = _ref9.nonce,
    nonce = _ref9$nonce === void 0 ? mockNonce() : _ref9$nonce;
  return {
    result: mockCreateTypedDataResult('CreateFollowBroadcastItemResult', {
      types: {
        Follow: [mockEIP712TypedDataField()]
      },
      domain: mockEIP712TypedDataDomain(),
      message: {
        nonce: nonce,
        deadline: 1644303500,
        followerProfileId: mockProfileId(),
        idsOfProfilesToFollow: [mockProfileId()],
        followTokenIds: [],
        datas: []
      }
    })
  };
}
function mockCreateFollowTypedDataResponse(_ref10) {
  var variables = _ref10.variables,
    data = _ref10.data;
  return {
    request: {
      query: CreateFollowTypedDataDocument,
      variables: variables
    },
    result: {
      data: data
    }
  };
}
function mockFollowResponse(_ref11) {
  var variables = _ref11.variables,
    data = _ref11.data;
  return {
    request: {
      query: FollowDocument,
      variables: variables
    },
    result: {
      data: data
    }
  };
}
function mockCreateUnfollowTypedDataData() {
  var _ref12 = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {},
    _ref12$nonce = _ref12.nonce,
    nonce = _ref12$nonce === void 0 ? mockNonce() : _ref12$nonce;
  return {
    result: mockCreateTypedDataResult('CreateUnfollowBroadcastItemResult', {
      types: {
        Unfollow: [mockEIP712TypedDataField()]
      },
      domain: mockEIP712TypedDataDomain(),
      message: {
        nonce: nonce,
        deadline: 1644303500,
        unfollowerProfileId: mockProfileId(),
        idsOfProfilesToUnfollow: [mockProfileId()]
      }
    })
  };
}
function mockCreateUnfollowTypedDataResponse(_ref13) {
  var variables = _ref13.variables,
    data = _ref13.data;
  return {
    request: {
      query: CreateUnfollowTypedDataDocument,
      variables: variables
    },
    result: {
      data: data
    }
  };
}
function mockUnfollowResponse(_ref14) {
  var variables = _ref14.variables,
    data = _ref14.data;
  return {
    request: {
      query: UnfollowDocument,
      variables: variables
    },
    result: {
      data: data
    }
  };
}
function mockCreateSetFollowModuleTypedDataData() {
  var _ref15 = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {},
    _ref15$nonce = _ref15.nonce,
    nonce = _ref15$nonce === void 0 ? mockNonce() : _ref15$nonce;
  return {
    result: mockCreateTypedDataResult('CreateSetFollowModuleBroadcastItemResult', {
      types: {
        SetFollowModule: [mockEIP712TypedDataField()]
      },
      domain: mockEIP712TypedDataDomain(),
      message: {
        nonce: nonce,
        deadline: 1644303500,
        profileId: mockProfileId(),
        followModule: mockEvmAddress(),
        followModuleInitData: '0x'
      }
    })
  };
}
function mockCreateSetFollowModuleTypedDataResponse(_ref16) {
  var variables = _ref16.variables,
    data = _ref16.data;
  return {
    request: {
      query: CreateSetFollowModuleTypedDataDocument,
      variables: variables
    },
    result: {
      data: data
    }
  };
}
function mockSetFollowModuleResponse(_ref17) {
  var variables = _ref17.variables,
    data = _ref17.data;
  return {
    request: {
      query: SetFollowModuleDocument,
      variables: variables
    },
    result: {
      data: data
    }
  };
}
function mockCreateLinkHandleToProfileTypedDataData() {
  var _ref18 = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {},
    _ref18$nonce = _ref18.nonce,
    nonce = _ref18$nonce === void 0 ? mockNonce() : _ref18$nonce;
  return {
    result: mockCreateTypedDataResult('CreateLinkHandleToProfileBroadcastItemResult', {
      types: {
        Link: [mockEIP712TypedDataField()]
      },
      domain: mockEIP712TypedDataDomain(),
      message: {
        nonce: nonce,
        deadline: 1644303500,
        profileId: mockProfileId(),
        handleId: mockEvmAddress()
      }
    })
  };
}
function mockCreateLinkHandleToProfileTypedDataResponse(_ref19) {
  var variables = _ref19.variables,
    data = _ref19.data;
  return {
    request: {
      query: CreateLinkHandleToProfileTypedDataDocument,
      variables: variables
    },
    result: {
      data: data
    }
  };
}
function mockLinkHandleToProfileResponse(_ref20) {
  var variables = _ref20.variables,
    data = _ref20.data;
  return {
    request: {
      query: LinkHandleToProfileDocument,
      variables: variables
    },
    result: {
      data: data
    }
  };
}
function mockCreateUnlinkHandleFromProfileTypedDataData() {
  var _ref21 = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {},
    _ref21$nonce = _ref21.nonce,
    nonce = _ref21$nonce === void 0 ? mockNonce() : _ref21$nonce;
  return {
    result: mockCreateTypedDataResult('CreateUnlinkHandleFromProfileBroadcastItemResult', {
      types: {
        Unlink: [mockEIP712TypedDataField()]
      },
      domain: mockEIP712TypedDataDomain(),
      message: {
        nonce: nonce,
        deadline: 1644303500,
        profileId: mockProfileId(),
        handleId: mockEvmAddress()
      }
    })
  };
}
function mockCreateUnlinkHandleFromProfileTypedDataResponse(_ref22) {
  var variables = _ref22.variables,
    data = _ref22.data;
  return {
    request: {
      query: CreateUnlinkHandleFromProfileTypedDataDocument,
      variables: variables
    },
    result: {
      data: data
    }
  };
}
function mockUnlinkHandleFromProfileResponse(_ref23) {
  var variables = _ref23.variables,
    data = _ref23.data;
  return {
    request: {
      query: UnlinkHandleFromProfileDocument,
      variables: variables
    },
    result: {
      data: data
    }
  };
}

function mockHidePublicationResponse(variables) {
  return {
    request: {
      query: HidePublicationDocument,
      variables: variables
    },
    result: {
      data: {
        hidePublication: null
      }
    }
  };
}
function mockReportPublicationResponse(args) {
  return {
    request: {
      query: ReportPublicationDocument,
      variables: args.variables
    },
    result: {
      data: {
        reportPublication: null
      }
    }
  };
}
function mockAddReactionResponse(_ref) {
  var variables = _ref.variables;
  return {
    request: {
      query: AddReactionDocument,
      variables: variables
    },
    result: {
      data: {
        addReaction: null
      }
    }
  };
}
function mockRemoveReactionResponse(_ref2) {
  var variables = _ref2.variables;
  return {
    request: {
      query: RemoveReactionDocument,
      variables: variables
    },
    result: {
      data: {
        removeReaction: null
      }
    }
  };
}
function mockCreateOnchainPostTypedDataData() {
  var _ref3 = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {},
    _ref3$nonce = _ref3.nonce,
    nonce = _ref3$nonce === void 0 ? mockNonce() : _ref3$nonce;
  return {
    result: mockCreateTypedDataResult('CreateOnchainPostBroadcastItemResult', {
      types: {
        Post: [mockEIP712TypedDataField()]
      },
      domain: mockEIP712TypedDataDomain(),
      message: {
        nonce: nonce,
        deadline: 1644303500,
        profileId: mockProfileId(),
        contentURI: 'ipfs://QmR5V6fwKWzoa9gevmYaQ11eMQsAahsjfWPz1rCoNJjN1K.json',
        actionModules: [mockEvmAddress()],
        actionModulesInitDatas: ['0x'],
        referenceModule: '0x0000000000000000000000000000000000000000',
        referenceModuleInitData: '0x'
      }
    })
  };
}
function mockCreateOnchainPostTypedDataResponse(_ref4) {
  var variables = _ref4.variables,
    data = _ref4.data;
  return {
    request: {
      query: CreateOnchainPostTypedDataDocument,
      variables: variables
    },
    result: {
      data: data
    }
  };
}
function mockPostOnchainResponse(_ref5) {
  var variables = _ref5.variables,
    data = _ref5.data;
  return {
    request: {
      query: PostOnchainDocument,
      variables: variables
    },
    result: {
      data: data
    }
  };
}
function mockCreateMomokaPostTypedDataData() {
  var _ref6 = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {},
    _ref6$nonce = _ref6.nonce,
    nonce = _ref6$nonce === void 0 ? mockNonce() : _ref6$nonce;
  return {
    result: mockCreateTypedDataResult('CreateMomokaPostBroadcastItemResult', {
      types: {
        Post: [mockEIP712TypedDataField()]
      },
      domain: mockEIP712TypedDataDomain(),
      message: {
        nonce: nonce,
        deadline: 1644303500,
        profileId: mockProfileId(),
        contentURI: 'ipfs://QmR5V6fwKWzoa9gevmYaQ11eMQsAahsjfWPz1rCoNJjN1K.json',
        actionModules: ['0xd6072BB2ABc0a9d1331c7d0B83AE6C47f2Cb86A3'],
        actionModulesInitDatas: ['0x'],
        referenceModule: '0x0000000000000000000000000000000000000000',
        referenceModuleInitData: '0x'
      }
    })
  };
}
function mockCreateMomokaPostTypedDataResponse(_ref7) {
  var variables = _ref7.variables,
    data = _ref7.data;
  return {
    request: {
      query: CreateMomokaPostTypedDataDocument,
      variables: variables
    },
    result: {
      data: data
    }
  };
}
function mockPostOnMomokaResponse(_ref8) {
  var variables = _ref8.variables,
    data = _ref8.data;
  return {
    request: {
      query: PostOnMomokaDocument,
      variables: variables
    },
    result: {
      data: data
    }
  };
}
function mockCreateMomokaPublicationResult() {
  return {
    __typename: 'CreateMomokaPublicationResult',
    id: mockPublicationId(),
    momokaId: faker.datatype.uuid(),
    proof: '...'
  };
}
function mockCreateOnchainCommentTypedDataData() {
  var _ref9 = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {},
    _ref9$nonce = _ref9.nonce,
    nonce = _ref9$nonce === void 0 ? mockNonce() : _ref9$nonce;
  return {
    result: mockCreateTypedDataResult('CreateOnchainCommentBroadcastItemResult', {
      types: {
        Comment: [mockEIP712TypedDataField()]
      },
      domain: mockEIP712TypedDataDomain(),
      message: {
        nonce: nonce,
        deadline: 1644303500,
        profileId: mockProfileId(),
        contentURI: 'ipfs://QmR5V6fwKWzoa9gevmYaQ11eMQsAahsjfWPz1rCoNJjN1K.json',
        pointedProfileId: mockProfileId(),
        pointedPubId: '0x01',
        referrerProfileIds: [],
        referrerPubIds: [],
        referenceModuleData: '0x',
        actionModules: [mockEvmAddress()],
        actionModulesInitDatas: ['0x'],
        referenceModule: '0x0000000000000000000000000000000000000000',
        referenceModuleInitData: '0x'
      }
    })
  };
}
function mockCreateOnchainCommentTypedDataResponse(_ref10) {
  var variables = _ref10.variables,
    data = _ref10.data;
  return {
    request: {
      query: CreateOnchainCommentTypedDataDocument,
      variables: variables
    },
    result: {
      data: data
    }
  };
}
function mockCommentOnchainResponse(_ref11) {
  var variables = _ref11.variables,
    data = _ref11.data;
  return {
    request: {
      query: CommentOnchainDocument,
      variables: variables
    },
    result: {
      data: data
    }
  };
}
function mockCreateMomokaCommentTypedDataData() {
  var _ref12 = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {},
    _ref12$nonce = _ref12.nonce,
    nonce = _ref12$nonce === void 0 ? mockNonce() : _ref12$nonce;
  return {
    result: mockCreateTypedDataResult('CreateMomokaCommentBroadcastItemResult', {
      types: {
        Comment: [mockEIP712TypedDataField()]
      },
      domain: mockEIP712TypedDataDomain(),
      message: {
        nonce: nonce,
        deadline: 1644303500,
        profileId: mockProfileId(),
        contentURI: 'ipfs://QmR5V6fwKWzoa9gevmYaQ11eMQsAahsjfWPz1rCoNJjN1K.json',
        pointedProfileId: mockProfileId(),
        pointedPubId: '0x01',
        referrerProfileIds: [],
        referrerPubIds: [],
        referenceModuleData: '0x',
        actionModules: [mockEvmAddress()],
        actionModulesInitDatas: ['0x'],
        referenceModule: '0x0000000000000000000000000000000000000000',
        referenceModuleInitData: '0x'
      }
    })
  };
}
function mockCreateMomokaCommentTypedDataResponse(_ref13) {
  var variables = _ref13.variables,
    data = _ref13.data;
  return {
    request: {
      query: CreateMomokaCommentTypedDataDocument,
      variables: variables
    },
    result: {
      data: data
    }
  };
}
function mockCommentOnMomokaResponse(_ref14) {
  var variables = _ref14.variables,
    data = _ref14.data;
  return {
    request: {
      query: CommentOnMomokaDocument,
      variables: variables
    },
    result: {
      data: data
    }
  };
}
function mockCreateMomokaMirrorTypedDataData() {
  var _ref15 = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {},
    _ref15$nonce = _ref15.nonce,
    nonce = _ref15$nonce === void 0 ? mockNonce() : _ref15$nonce;
  return {
    result: mockCreateTypedDataResult('CreateMomokaMirrorBroadcastItemResult', {
      types: {
        Mirror: [mockEIP712TypedDataField()]
      },
      domain: mockEIP712TypedDataDomain(),
      message: {
        nonce: nonce,
        deadline: 1644303500,
        profileId: mockProfileId(),
        metadataURI: 'ipfs://QmR5V6fwKWzoa9gevmYaQ11eMQsAahsjfWPz1rCoNJjN1K.json',
        pointedProfileId: mockProfileId(),
        pointedPubId: '0x01',
        referrerProfileIds: [],
        referrerPubIds: [],
        referenceModuleData: '0x'
      }
    })
  };
}
function mockCreateMomokaMirrorTypedDataResponse(_ref16) {
  var variables = _ref16.variables,
    data = _ref16.data;
  return {
    request: {
      query: CreateMomokaMirrorTypedDataDocument,
      variables: variables
    },
    result: {
      data: data
    }
  };
}
function mockMirrorOnMomokaResponse(_ref17) {
  var variables = _ref17.variables,
    data = _ref17.data;
  return {
    request: {
      query: MirrorOnMomokaDocument,
      variables: variables
    },
    result: {
      data: data
    }
  };
}
function mockCreateOnchainMirrorTypedDataData() {
  var _ref18 = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {},
    _ref18$nonce = _ref18.nonce,
    nonce = _ref18$nonce === void 0 ? mockNonce() : _ref18$nonce;
  return {
    result: mockCreateTypedDataResult('CreateOnchainMirrorBroadcastItemResult', {
      types: {
        Mirror: [mockEIP712TypedDataField()]
      },
      domain: mockEIP712TypedDataDomain(),
      message: {
        nonce: nonce,
        deadline: 1644303500,
        profileId: mockProfileId(),
        metadataURI: 'ipfs://QmR5V6fwKWzoa9gevmYaQ11eMQsAahsjfWPz1rCoNJjN1K.json',
        pointedProfileId: mockProfileId(),
        pointedPubId: '0x01',
        referrerProfileIds: [],
        referrerPubIds: [],
        referenceModuleData: '0x'
      }
    })
  };
}
function mockCreateOnchainMirrorTypedDataResponse(_ref19) {
  var variables = _ref19.variables,
    data = _ref19.data;
  return {
    request: {
      query: CreateOnchainMirrorTypedDataDocument,
      variables: variables
    },
    result: {
      data: data
    }
  };
}
function mockMirrorOnchainResponse(_ref20) {
  var variables = _ref20.variables,
    data = _ref20.data;
  return {
    request: {
      query: MirrorOnchainDocument,
      variables: variables
    },
    result: {
      data: data
    }
  };
}
function mockCreateOnchainQuoteTypedDataData() {
  var _ref21 = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {},
    _ref21$nonce = _ref21.nonce,
    nonce = _ref21$nonce === void 0 ? mockNonce() : _ref21$nonce;
  return {
    result: mockCreateTypedDataResult('CreateOnchainQuoteBroadcastItemResult', {
      types: {
        Quote: [mockEIP712TypedDataField()]
      },
      domain: mockEIP712TypedDataDomain(),
      message: {
        nonce: nonce,
        deadline: 1644303500,
        profileId: mockProfileId(),
        contentURI: 'ipfs://QmR5V6fwKWzoa9gevmYaQ11eMQsAahsjfWPz1rCoNJjN1K.json',
        pointedProfileId: mockProfileId(),
        pointedPubId: '0x01',
        referrerProfileIds: [],
        referrerPubIds: [],
        referenceModuleData: '0x',
        actionModules: [mockEvmAddress()],
        actionModulesInitDatas: ['0x'],
        referenceModule: '0x0000000000000000000000000000000000000000',
        referenceModuleInitData: '0x'
      }
    })
  };
}
function mockCreateOnchainQuoteTypedDataResponse(_ref22) {
  var variables = _ref22.variables,
    data = _ref22.data;
  return {
    request: {
      query: CreateOnchainQuoteTypedDataDocument,
      variables: variables
    },
    result: {
      data: data
    }
  };
}
function mockQuoteOnchainResponse(_ref23) {
  var variables = _ref23.variables,
    data = _ref23.data;
  return {
    request: {
      query: QuoteOnchainDocument,
      variables: variables
    },
    result: {
      data: data
    }
  };
}
function mockCreateMomokaQuoteTypedDataData() {
  var _ref24 = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {},
    _ref24$nonce = _ref24.nonce,
    nonce = _ref24$nonce === void 0 ? mockNonce() : _ref24$nonce;
  return {
    result: mockCreateTypedDataResult('CreateMomokaQuoteBroadcastItemResult', {
      types: {
        Quote: [mockEIP712TypedDataField()]
      },
      domain: mockEIP712TypedDataDomain(),
      message: {
        nonce: nonce,
        deadline: 1644303500,
        profileId: mockProfileId(),
        contentURI: 'ipfs://QmR5V6fwKWzoa9gevmYaQ11eMQsAahsjfWPz1rCoNJjN1K.json',
        pointedProfileId: mockProfileId(),
        pointedPubId: '0x01',
        referrerProfileIds: [],
        referrerPubIds: [],
        referenceModuleData: '0x',
        actionModules: [mockEvmAddress()],
        actionModulesInitDatas: ['0x'],
        referenceModule: '0x0000000000000000000000000000000000000000',
        referenceModuleInitData: '0x'
      }
    })
  };
}
function mockCreateMomokaQuoteTypedDataResponse(_ref25) {
  var variables = _ref25.variables,
    data = _ref25.data;
  return {
    request: {
      query: CreateMomokaQuoteTypedDataDocument,
      variables: variables
    },
    result: {
      data: data
    }
  };
}
function mockQuoteOnMomokaResponse(_ref26) {
  var variables = _ref26.variables,
    data = _ref26.data;
  return {
    request: {
      query: QuoteOnMomokaDocument,
      variables: variables
    },
    result: {
      data: data
    }
  };
}
function mockAddToMyBookmarksResponse(variables) {
  return {
    request: {
      query: AddPublicationBookmarkDocument,
      variables: variables
    },
    result: {
      data: {
        result: null
      }
    }
  };
}
function mockRemoveFromMyBookmarksResponse(variables) {
  return {
    request: {
      query: RemovePublicationBookmarkDocument,
      variables: variables
    },
    result: {
      data: {
        result: null
      }
    }
  };
}
function mockCreateLegacyCollectTypedDataData() {
  var _ref27 = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {},
    _ref27$nonce = _ref27.nonce,
    nonce = _ref27$nonce === void 0 ? mockNonce() : _ref27$nonce;
  return {
    result: mockCreateTypedDataResult('CreateLegacyCollectBroadcastItemResult', {
      types: {
        CollectLegacy: [mockEIP712TypedDataField()]
      },
      domain: mockEIP712TypedDataDomain(),
      message: {
        nonce: nonce,
        deadline: 1644303500,
        publicationCollectedProfileId: mockProfileId(),
        publicationCollectedId: '0x01',
        collectorProfileId: mockProfileId(),
        referrerProfileId: mockProfileId(),
        referrerPubId: faker.datatype.hexadecimal({
          length: 2
        }),
        collectModuleData: '0x'
      }
    })
  };
}
function mockCreateLegacyCollectTypedDataResponse(_ref28) {
  var variables = _ref28.variables,
    data = _ref28.data;
  return {
    request: {
      query: CreateLegacyCollectTypedDataDocument,
      variables: variables
    },
    result: {
      data: data
    }
  };
}
function mockLegacyCollectResponse(_ref29) {
  var variables = _ref29.variables,
    data = _ref29.data;
  return {
    request: {
      query: LegacyCollectDocument,
      variables: variables
    },
    result: {
      data: data
    }
  };
}
function mockAddPublicationNotInterestedResponse(variables) {
  return {
    request: {
      query: AddPublicationNotInterestedDocument,
      variables: variables
    },
    result: {
      data: {
        addPublicationNotInterested: null
      }
    }
  };
}
function mockUndoPublicationNotInterestedResponse(variables) {
  return {
    request: {
      query: UndoPublicationNotInterestedDocument,
      variables: variables
    },
    result: {
      data: {
        undoPublicationNotInterested: null
      }
    }
  };
}

function mockBroadcastOnchainData(result) {
  return {
    result: result
  };
}
function mockBroadcastOnchainResponse(_ref) {
  var result = _ref.result,
    variables = _ref.variables;
  return {
    request: {
      query: BroadcastOnchainDocument,
      variables: variables
    },
    result: {
      data: mockBroadcastOnchainData(result)
    }
  };
}
function mockRelaySuccessFragment() {
  var txHash = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : mockTransactionHash();
  return {
    __typename: 'RelaySuccess',
    txHash: txHash,
    txId: faker.datatype.uuid()
  };
}
function mockRelayErrorFragment(reason) {
  return {
    __typename: 'RelayError',
    reason: reason
  };
}
function mockLensProfileManagerRelayError(reason) {
  return {
    __typename: 'LensProfileManagerRelayError',
    reason: reason
  };
}
function mockBroadcastOnMomokaResponse(_ref2) {
  var variables = _ref2.variables,
    result = _ref2.result;
  return {
    request: {
      query: BroadcastOnMomokaDocument,
      variables: variables
    },
    result: {
      data: {
        result: result
      }
    }
  };
}

/**
 * Mock any response by matching the query and variables.
 */
function mockAnyResponse(bulk) {
  return {
    request: {
      query: bulk.request.query
    },
    variableMatcher: function variableMatcher(variables) {
      if (!bulk.request.variables) {
        return true;
      }
      try {
        expect(variables).toMatchObject(bulk.request.variables);
        return true;
      } catch (_unused) {
        return false;
      }
    },
    result: bulk.result
  };
}

/**
 * Mock any paginated responses by matching the query and variables.
 */
function mockAnyPaginatedResponse(_ref) {
  var variables = _ref.variables,
    items = _ref.items,
    _ref$info = _ref.info,
    info = _ref$info === void 0 ? mockPaginatedResultInfo() : _ref$info,
    query = _ref.query;
  return mockAnyResponse({
    request: {
      query: query,
      variables: variables
    },
    result: {
      data: {
        result: {
          items: items,
          pageInfo: info
        }
      }
    }
  });
}

function mockFeedResponse(_ref) {
  var variables = _ref.variables,
    items = _ref.items;
  return mockAnyPaginatedResponse({
    variables: variables,
    items: items,
    query: FeedDocument
  });
}
function mockExplorePublicationsResponse(args) {
  return mockAnyPaginatedResponse({
    variables: args.variables,
    items: args.items,
    query: ExplorePublicationsDocument
  });
}
function mockFeedHighlightsResponse(_ref2) {
  var variables = _ref2.variables,
    items = _ref2.items;
  return mockAnyPaginatedResponse({
    variables: variables,
    items: items,
    query: FeedHighlightsDocument
  });
}
function mockExploreProfilesResponse(_ref3) {
  var variables = _ref3.variables,
    items = _ref3.items;
  return mockAnyPaginatedResponse({
    variables: variables,
    items: items,
    query: ExploreProfilesDocument
  });
}

function mockCurrenciesResponse(_ref) {
  var variables = _ref.variables,
    items = _ref.items,
    _ref$info = _ref.info,
    info = _ref$info === void 0 ? mockPaginatedResultInfo() : _ref$info;
  return {
    request: {
      query: CurrenciesDocument,
      variables: variables
    },
    result: {
      data: {
        result: {
          items: items.map(function (currency) {
            return {
              __typename: 'Erc20',
              name: currency.name,
              symbol: currency.symbol,
              decimals: currency.decimals,
              contract: {
                __typename: 'NetworkAddress',
                address: currency.address,
                chainId: 80002
              }
            };
          }),
          pageInfo: info
        }
      }
    }
  };
}

function mockModuleMetadataResponse(_ref) {
  var implementation = _ref.implementation,
    result = _ref.result;
  return {
    request: {
      query: ModuleMetadataDocument,
      variables: {
        request: {
          implementation: implementation
        }
      }
    },
    result: {
      data: {
        result: result
      }
    }
  };
}

function mockNotificationsResponse(_ref) {
  var variables = _ref.variables,
    items = _ref.items,
    _ref$info = _ref.info,
    info = _ref$info === void 0 ? mockPaginatedResultInfo() : _ref$info;
  return mockAnyPaginatedResponse({
    variables: variables,
    items: items,
    query: NotificationsDocument,
    info: info
  });
}

function mockProfileResponse(_ref) {
  var variables = _ref.variables,
    result = _ref.result;
  return mockAnyResponse({
    request: {
      query: ProfileDocument,
      variables: variables
    },
    result: {
      data: {
        result: result
      }
    }
  });
}
function mockProfilesResponse(_ref2) {
  var variables = _ref2.variables,
    items = _ref2.items,
    _ref2$info = _ref2.info,
    info = _ref2$info === void 0 ? mockPaginatedResultInfo() : _ref2$info;
  return mockAnyPaginatedResponse({
    variables: variables,
    items: items,
    info: info,
    query: ProfilesDocument
  });
}
function mockMutualFollowersResponse(_ref3) {
  var variables = _ref3.variables,
    items = _ref3.items,
    _ref3$info = _ref3.info,
    info = _ref3$info === void 0 ? mockPaginatedResultInfo() : _ref3$info;
  return mockAnyPaginatedResponse({
    variables: variables,
    items: items,
    info: info,
    query: MutualFollowersDocument
  });
}
function mockFollowersResponse(_ref4) {
  var variables = _ref4.variables,
    items = _ref4.items,
    _ref4$info = _ref4.info,
    info = _ref4$info === void 0 ? mockPaginatedResultInfo() : _ref4$info;
  return mockAnyPaginatedResponse({
    variables: variables,
    items: items,
    info: info,
    query: FollowersDocument
  });
}
function mockFollowingResponse(_ref5) {
  var variables = _ref5.variables,
    items = _ref5.items,
    _ref5$info = _ref5.info,
    info = _ref5$info === void 0 ? mockPaginatedResultInfo() : _ref5$info;
  return mockAnyPaginatedResponse({
    variables: variables,
    items: items,
    info: info,
    query: FollowingDocument
  });
}
function mockSearchProfilesResponse(_ref6) {
  var variables = _ref6.variables,
    items = _ref6.items,
    _ref6$info = _ref6.info,
    info = _ref6$info === void 0 ? mockPaginatedResultInfo() : _ref6$info;
  return mockAnyPaginatedResponse({
    variables: variables,
    items: items,
    info: info,
    query: SearchProfilesDocument
  });
}
function mockProfileRecommendationsResponse(_ref7) {
  var variables = _ref7.variables,
    items = _ref7.items,
    _ref7$info = _ref7.info,
    info = _ref7$info === void 0 ? mockPaginatedResultInfo() : _ref7$info;
  return mockAnyPaginatedResponse({
    variables: variables,
    items: items,
    info: info,
    query: ProfileRecommendationsDocument
  });
}
function mockWhoActedOnPublicationResponse(_ref8) {
  var variables = _ref8.variables,
    items = _ref8.items,
    _ref8$info = _ref8.info,
    info = _ref8$info === void 0 ? mockPaginatedResultInfo() : _ref8$info;
  return mockAnyPaginatedResponse({
    variables: variables,
    items: items,
    info: info,
    query: WhoActedOnPublicationDocument
  });
}
function mockProfileActionHistoryResponse(_ref9) {
  var variables = _ref9.variables,
    items = _ref9.items,
    _ref9$info = _ref9.info,
    info = _ref9$info === void 0 ? mockPaginatedResultInfo() : _ref9$info;
  return {
    request: {
      query: ProfileActionHistoryDocument,
      variables: variables
    },
    result: {
      data: {
        result: {
          items: items,
          pageInfo: info
        }
      }
    }
  };
}
function mockWhoReactedToPublicationResponse(_ref10) {
  var variables = _ref10.variables,
    items = _ref10.items,
    _ref10$info = _ref10.info,
    info = _ref10$info === void 0 ? mockPaginatedResultInfo() : _ref10$info;
  return mockAnyPaginatedResponse({
    variables: variables,
    items: items,
    info: info,
    query: WhoReactedPublicationDocument
  });
}

function mockPublicationResponse(_ref) {
  var variables = _ref.variables,
    result = _ref.result;
  return mockAnyResponse({
    request: {
      query: PublicationDocument,
      variables: variables
    },
    result: {
      data: {
        result: result
      }
    }
  });
}
function mockPublicationsResponse(_ref2) {
  var variables = _ref2.variables,
    items = _ref2.items,
    _ref2$info = _ref2.info,
    info = _ref2$info === void 0 ? mockPaginatedResultInfo() : _ref2$info;
  return mockAnyPaginatedResponse({
    variables: variables,
    items: items,
    info: info,
    query: PublicationsDocument
  });
}
function mockSearchPublicationsResponse(_ref3) {
  var variables = _ref3.variables,
    items = _ref3.items,
    _ref3$info = _ref3.info,
    info = _ref3$info === void 0 ? mockPaginatedResultInfo() : _ref3$info;
  return mockAnyPaginatedResponse({
    variables: variables,
    items: items,
    info: info,
    query: SearchPublicationsDocument
  });
}
function mockProfileBookmarksResponse(_ref4) {
  var variables = _ref4.variables,
    items = _ref4.items,
    _ref4$info = _ref4.info,
    info = _ref4$info === void 0 ? mockPaginatedResultInfo() : _ref4$info;
  return mockAnyPaginatedResponse({
    variables: variables,
    items: items,
    info: info,
    query: PublicationBookmarksDocument
  });
}

function mockFollowRevenuesResponse(_ref) {
  var variables = _ref.variables,
    result = _ref.result;
  return mockAnyResponse({
    request: {
      query: FollowRevenuesDocument,
      variables: variables
    },
    result: {
      data: {
        result: {
          revenues: result
        }
      }
    }
  });
}
function mockRevenueFromPublicationResponse(_ref2) {
  var variables = _ref2.variables,
    result = _ref2.result;
  return mockAnyResponse({
    request: {
      query: RevenueFromPublicationDocument,
      variables: variables
    },
    result: {
      data: {
        result: result
      }
    }
  });
}
function mockRevenueFromPublicationsResponse(_ref3) {
  var variables = _ref3.variables,
    items = _ref3.items,
    _ref3$info = _ref3.info,
    info = _ref3$info === void 0 ? mockPaginatedResultInfo() : _ref3$info;
  return mockAnyPaginatedResponse({
    variables: variables,
    items: items,
    info: info,
    query: RevenueFromPublicationsDocument
  });
}

function mockLensTransactionStatusDataResponse(_ref) {
  var request = _ref.request,
    result = _ref.result;
  return {
    request: {
      query: LensTransactionStatusDocument,
      variables: {
        request: request
      }
    },
    result: {
      data: {
        result: result
      }
    }
  };
}
function mockLensTransactionResult(overrides) {
  return _objectSpread2(_objectSpread2({
    extraInfo: null,
    reason: null,
    status: LensTransactionStatusType.Processing,
    txHash: mockTransactionHash()
  }, overrides), {}, {
    __typename: 'LensTransactionResult'
  });
}
function mockGenerateLensAPIRelayAddressResponse(_ref2) {
  var address = _ref2.address;
  return {
    request: {
      query: GenerateLensApiRelayAddressDocument
    },
    result: {
      data: {
        result: address
      }
    }
  };
}

function mockCursor() {
  return faker.random.alphaNumeric(10);
}

export { AnyPaginatedQueryDocument, createHttpJsonResponse, createUnauthenticatedHttpResponse, mockActOnOpenActionResponse, mockAddPublicationNotInterestedResponse, mockAddReactionResponse, mockAddToMyBookmarksResponse, mockAmountFragment, mockAmountFragmentFrom, mockAnyPaginatedQueryResponse, mockAnyPaginatedQueryResult, mockBlockProfilesResponse, mockBlockProfilesTypedDataData, mockBlockProfilesTypedDataResponse, mockBroadcastOnMomokaResponse, mockBroadcastOnchainResponse, mockCommentFragment, mockCommentOnMomokaResponse, mockCommentOnchainResponse, mockCreateActOnOpenActionTypedDataData, mockCreateActOnOpenActionTypedDataResponse, mockCreateFollowTypedDataData, mockCreateFollowTypedDataResponse, mockCreateLegacyCollectTypedDataData, mockCreateLegacyCollectTypedDataResponse, mockCreateLinkHandleToProfileTypedDataData, mockCreateLinkHandleToProfileTypedDataResponse, mockCreateMomokaCommentTypedDataData, mockCreateMomokaCommentTypedDataResponse, mockCreateMomokaMirrorTypedDataData, mockCreateMomokaMirrorTypedDataResponse, mockCreateMomokaPostTypedDataData, mockCreateMomokaPostTypedDataResponse, mockCreateMomokaPublicationResult, mockCreateMomokaQuoteTypedDataData, mockCreateMomokaQuoteTypedDataResponse, mockCreateOnchainCommentTypedDataData, mockCreateOnchainCommentTypedDataResponse, mockCreateOnchainMirrorTypedDataData, mockCreateOnchainMirrorTypedDataResponse, mockCreateOnchainPostTypedDataData, mockCreateOnchainPostTypedDataResponse, mockCreateOnchainQuoteTypedDataData, mockCreateOnchainQuoteTypedDataResponse, mockCreateSetFollowModuleTypedDataData, mockCreateSetFollowModuleTypedDataResponse, mockCreateSetProfileMetadataTypedDataData, mockCreateSetProfileMetadataTypedDataResponse, mockCreateUnfollowTypedDataData, mockCreateUnfollowTypedDataResponse, mockCreateUnlinkHandleFromProfileTypedDataData, mockCreateUnlinkHandleFromProfileTypedDataResponse, mockCurrenciesResponse, mockCursor, mockErc20Fragment, mockExploreProfilesResponse, mockExplorePublicationsResponse, mockFeeFollowModuleSettingsFragment, mockFeedHighlightsResponse, mockFeedItemFragment, mockFeedResponse, mockFollowResponse, mockFollowRevenuesResponse, mockFollowersResponse, mockFollowingResponse, mockGenerateLensAPIRelayAddressResponse, mockGenericErrorResponse, mockGenericSuccessResponse, mockHandleInfo, mockHidePublicationResponse, mockLegacyAaveFeeCollectModuleSettingsFragment, mockLegacyCollectResponse, mockLegacyErc4626FeeCollectModuleSettingsFragment, mockLegacyFeeCollectModuleSettingsFragment, mockLegacyFreeCollectModuleSettingsFragment, mockLegacyLimitedFeeCollectModuleSettingsFragment, mockLegacyLimitedTimedFeeCollectModuleSettingsFragment, mockLegacyMultirecipientFeeCollectModuleSettingsFragment, mockLegacyRevertCollectModuleSettingsFragment, mockLegacySimpleCollectModuleSettingsFragment, mockLegacyTimedFeeCollectModuleSettingsFragment, mockLensApolloClient, mockLensCache, mockLensProfileManagerRelayError, mockLensTransactionResult, mockLensTransactionStatusDataResponse, mockLinkHandleToProfileResponse, mockMentionNotification, mockMirrorFragment, mockMirrorOnMomokaResponse, mockMirrorOnchainResponse, mockModuleMetadataFragment, mockModuleMetadataResponse, mockModuleMetadataResultFragment, mockMultirecipientFeeCollectOpenActionSettingsFragment, mockMutualFollowersResponse, mockNetworkAddressFragment, mockNotificationsResponse, mockPaginatedResultInfo, mockPostFragment, mockPostOnMomokaResponse, mockPostOnchainResponse, mockProfileActionHistoryFragment, mockProfileActionHistoryResponse, mockProfileBookmarksResponse, mockProfileFragment, mockProfileReactionResultFragment, mockProfileRecommendationsResponse, mockProfileResponse, mockProfileStatsFragment, mockProfileWhoReactedResultFragment, mockProfilesResponse, mockPublicationOperationsFragment, mockPublicationResponse, mockPublicationRevenueFragment, mockPublicationStatsFragment, mockPublicationTextOnlyMetadata, mockPublicationsResponse, mockQuoteFragment, mockQuoteOnMomokaResponse, mockQuoteOnchainResponse, mockRelayErrorFragment, mockRelaySuccessFragment, mockRemoveFromMyBookmarksResponse, mockRemoveReactionResponse, mockReportPublicationResponse, mockRevenueAggregateFragment, mockRevenueFromPublicationResponse, mockRevenueFromPublicationsResponse, mockRevokeAuthenticationResponse, mockSearchProfilesResponse, mockSearchPublicationsResponse, mockSetFollowModuleResponse, mockSimpleCollectOpenActionSettingsFragment, mockTransactionState, mockUnblockProfilesResponse, mockUnblockProfilesTypedDataData, mockUnblockProfilesTypedDataResponse, mockUndoPublicationNotInterestedResponse, mockUnfollowResponse, mockUnknownOpenActionModuleSettingsFragment, mockUnknownReferenceModuleSettings, mockUnlinkHandleFromProfileResponse, mockValidationErrorResponse, mockWhoActedOnPublicationResponse, mockWhoReactedToPublicationResponse };
