// @lens-protocol/domain/dist/ReportClaim-2d81b360.esm.js
import { a as _createClass, _ as _classCallCheck } from './createClass-2012b414.esm.js';

var Profile = /*#__PURE__*/function () {
  function Profile(id, handle) {
    _classCallCheck(this, Profile);
    this.id = id;
    this.handle = handle;
  }
  _createClass(Profile, null, [{
    key: "create",
    value: function create(_ref) {
      var id = _ref.id,
        handle = _ref.handle;
      return new Profile(id, handle);
    }
  }]);
  return Profile;
}();

// No entity was required so far for this use case.

var ReportReason = /*#__PURE__*/function (ReportReason) {
  ReportReason["ANIMAL_ABUSE"] = "ANIMAL_ABUSE";
  ReportReason["HARASSMENT"] = "HARASSMENT";
  ReportReason["VIOLENCE"] = "VIOLENCE";
  ReportReason["SELF_HARM"] = "SELF_HARM";
  ReportReason["DIRECT_THREAT"] = "DIRECT_THREAT";
  ReportReason["HATE_SPEECH"] = "HATE_SPEECH";
  ReportReason["NUDITY"] = "NUDITY";
  ReportReason["OFFENSIVE"] = "OFFENSIVE";
  ReportReason["SCAM"] = "SCAM";
  ReportReason["UNAUTHORIZED_SALE"] = "UNAUTHORIZED_SALE";
  ReportReason["IMPERSONATION"] = "IMPERSONATION";
  ReportReason["MISLEADING"] = "MISLEADING";
  ReportReason["MISUSE_HASHTAGS"] = "MISUSE_HASHTAGS";
  ReportReason["UNRELATED"] = "UNRELATED";
  ReportReason["REPETITIVE"] = "REPETITIVE";
  ReportReason["FAKE_ENGAGEMENT"] = "FAKE_ENGAGEMENT";
  ReportReason["MANIPULATION_ALGO"] = "MANIPULATION_ALGO";
  ReportReason["SOMETHING_ELSE"] = "SOMETHING_ELSE";
  return ReportReason;
}({});

export { Profile as P, ReportReason as R };
